providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/model/FlagParser.java

```diff
@@ -40,37 +39,39 @@ private FlagParser() {
     }
 
     static {
-        try (InputStream schema = FlagParser.class.getClassLoader().getResourceAsStream(SCHEMA_RESOURCE)) {
-            if (schema == null) {
-                log.warn(String.format("Resource %s not found", SCHEMA_RESOURCE));
-            } else {
-                final ByteArrayOutputStream result = new ByteArrayOutputStream();
-                byte[] buffer = new byte[512];
-                for (int size; 0 < (size = schema.read(buffer)); ) {
-                    result.write(buffer, 0, size);
-                }
-
-                JsonSchemaFactory instance = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
-                SCHEMA_VALIDATOR = instance.getSchema(result.toString("UTF-8"));
-            }
+        try {
+            // load both schemas from resources (root (flags.json) and referenced (targeting.json)
+            // we don't want to resolve anything from the network
+            Map<String, String> mappings = new HashMap<>();
+            mappings.put("https://flagd.dev/schema/v0/targeting.json", "classpath:targeting.json");
+            mappings.put("https://flagd.dev/schema/v0/flags.json", "classpath:flags.json");
+
+            SCHEMA_VALIDATOR = JsonSchemaFactory
+                    .getInstance(SpecVersion.VersionFlag.V7,
+                            builder -> builder
+                                    .schemaMappers(schemaMappers -> schemaMappers.mappings(mappings)))
+                    .getSchema(new URI("https://flagd.dev/schema/v0/flags.json"));
         } catch (Throwable e) {
             // log only, do not throw
-            log.warn(String.format("Error loading resource %s, schema validation will be skipped", SCHEMA_RESOURCE), e);
+            log.warn(String.format("Error loading schema schema resources, schema validation will be skipped"));
         }
     }
 
     /**
      * Parse {@link String} for feature flags.
      */
-    public static Map<String, FeatureFlag> parseString(final String configuration) throws IOException {
+    public static Map<String, FeatureFlag> parseString(final String configuration, boolean throwIfInvalid)
+            throws IOException {
         if (SCHEMA_VALIDATOR != null) {
             try (JsonParser parser = MAPPER.createParser(configuration)) {
                 Set<ValidationMessage> validationMessages = SCHEMA_VALIDATOR.validate(parser.readValueAsTree());
 
                 if (!validationMessages.isEmpty()) {
-                    throw new IllegalArgumentException(
-                            String.format("Failed to parse configurations. %d validation error(s) reported.",
-                                    validationMessages.size()));
+                    String message = String.format("Invalid flag configuration: %s", validationMessages.toArray());
+                    log.warn(message);
+                    if (throwIfInvalid) {
+                        throw new IllegalArgumentException(message);
+                    }
```

Comment:
I no longer throw if things are invalid... instead just log. This is consistent with flagd itself, but I'm happy to talk about syncing this behavior across providers/flagd. Interested in what others think. I've left all the validation in, and in fact it can be enabled to throw instead of warn with a single bool.

Reply:
I think this is fine. Parsing will anyway fail if we cannot derive `FeatureFlag` instances. I think it's the targeting rules that can get impacted if something's wrong.

Reply:
I think having this behavior configurable is the most sensible decision :)
I wonder now if we should always throw here and swallow the exception in FlagStore 🤔
I would suggest to throw here because otherwise, to know if the parsing was successful, we need to look into the size of the map 🤔

Reply:
We should not swallow the error and instead at least log. This is what we do here as the default behavior. 

Inside flag store, exception will make the provider to emit a stale state [1]. 

[1] - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStore.java#L110

Reply:
@thisthat 

The targeting doesn't impact the size of the map - it's not used at all until the flag is actually evaluated and it's validity doesn't prevent the flags from being parsed into the map. The behavior in this PR is consistent with flagd itself: it simply warns the targeting is invalid and logs it, and when the expression is evaluated it defaults (or if it can't be parsed at all it errors).

I'm happy to discuss this further in a separate issue if you still have concerns though.

Reply:
@toddbaert Yes, this is how I see this. Earlier, we would have failed on targeting rule evaluations and emit a stale state. But now, instead we store this flag and fail on the evaluation. Given this is consistent with flagd,  personally I think this is fine :)